netsim/modules/lag.py

```diff
@@ -0,0 +1,99 @@
+import typing
+import netaddr
+import re
+
+from box import Box, BoxList
+from . import _Module, _dataplane
+from .. import data
+from ..utils import log
+from ..augment import devices, links
+
+GROUPNAME = re.compile(r'^links\[(?P<group>[\w\-_]+)\]\[\d+\]')
+
+ID_SET = 'lag_id'
+
+def populate_lag_id_set(topology: Box) -> None:
+  _dataplane.create_id_set(ID_SET)
+  LAG_IDS = { l.lag.ifindex for l in topology.links if 'lag' in l and 'ifindex' in l.lag }
+  _dataplane.extend_id_set(ID_SET,LAG_IDS)
+  _dataplane.set_id_counter(ID_SET,topology.defaults.lag.start_lag_id,100)
+
+class LAG(_Module):
+
+  def module_pre_transform(self, topology: Box) -> None:
+    populate_lag_id_set(topology)
+
+  """
+  link_pre_transform: Process LAG link groups and add virtual parent links to the topology
+  """
+  def link_pre_transform(self, link: Box, topology: Box) -> None:
+    if log.debug_active('lag'):
+      print(f'LAG link_pre_transform for {link}')
+
+    # Iterate over links with type lag, created for link group(s)
+    if link.get('type',"")=="lag" and not ('lag' in link and '_parent' in link.lag):
+      _match = GROUPNAME.search(link._linkname)
```

Comment:
We could also have the link group code add an internal ```group``` attribute to the links it creates, instead of reverse engineering the link names

Reply:
That would make much more sense than reverse-engineering. Just set '_link_group' attribute in copy_group_data somewhere around https://github.com/ipspace/netlab/blob/dev/netsim/augment/links.py#L1057. It would also be nice to check if the parent already has '_link_group' attribute and make hierarchical group names if necessary.